tools/jnimarshalmethod-gen/App.cs

```diff
@@ -526,9 +522,12 @@ static void ColorMessage (string message, ConsoleColor color, TextWriter writer,
 
 		public static void ColorWrite (string message, ConsoleColor color) => ColorMessage (message, color, Console.Out, false);
 
-		public static void Error (string message) => ColorMessage ($"Error: {Name}: {message}", ConsoleColor.Red, Console.Error);
+		public static void ErrorAndExit (Message message, int code, params object[] args) {
```

Comment:
Currently, these do not seem to follow the MSBuild error/warning format, and thus will not be reported to users in the IDE.  

If we intend for them to be MSBuild errors/warnings, then we need to change them to meet the required format:
https://github.com/dotnet/msbuild/blob/master/src/Shared/CanonicalError.cs

If we do not intend for them to be MSBuild warnings/errors, then we do not need to localize them at this time.

Reply:
Does it not? Which part is wrong? The error code is optional, so I guess the message should make it to the IDE.

Reply:
Interesting, maybe it will.  According to this the error code is required: https://docs.microsoft.com/en-us/archive/blogs/msbuild/msbuild-visual-studio-aware-error-messages-and-message-formats, but https://github.com/dotnet/msbuild/blob/master/src/Shared/CanonicalError.cs lists it as optional.

I think we still may want to add the code in.  Once these are translated, people won't have something to search for, since the message will be localized.

Reply:
Thanks for the second link. I think I will add the codes then to be sure and to make it easier to find related content as you suggested.

Reply:
In case it's handy to have it right here for reference, at the moment, with `jnimarshalmethod-gen.exe` being run via the `<Exec/>` task, the errors would appear in the IDE like:

![image](https://user-images.githubusercontent.com/4934473/91213470-a3e4d580-e6df-11ea-9bb0-5d8959373ea3.png)
![image](https://user-images.githubusercontent.com/4934473/91213733-063dd600-e6e0-11ea-8891-89e4cdb2f87e.png)

Based on a quick sanity-check with a test console app that just prints out a warning, the warnings behave similarly.

So adding error codes and localizing those messages sounds good to me.

For comparison in case anyone reading the PR might be curious, the story for `generator.exe` (in particular the `ApiXmlAdjuster` log messages) is slightly different. That executable is run by a custom `<Generator/>` task that [uses xamarin-android's `LogFromStandardError()`](https://github.com/xamarin/xamarin-android/blob/f14890fa2146913afb023a37b6ff4ec62381392a/src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs#L185) extension method to match log output. That method currently has stricter requirements to match an error or warning, such as requiring `error` or `warning` to be lowercase: https://github.com/xamarin/xamarin-android/blob/f14890fa2146913afb023a37b6ff4ec62381392a/src/Xamarin.Android.Build.Tasks/Utilities/MSBuildExtensions.cs#L78-L79

Reply:
Thanks for getting more info about it. We should now print the codes after lastest commit added to this PR.